pkg/ingester/ingester.go

```diff
@@ -263,8 +272,11 @@ func (i *Ingester) Shutdown() {
 		close(i.quit)
 		i.done.Wait()
 
-		// Next initiate our graceful exit from the ring.
-		i.lifecycler.Shutdown()
+		if !i.cfg.FlushMode {
+			// Next initiate our graceful exit from the ring.
+			// If ingester was in flush mode, then it would not be added in the ring.
+			i.lifecycler.Shutdown()
```

Comment:
I wonder what happens if we call `i.lifecycler.Shutdown()` anyways?

Reply:
nil pointer exception. `i.lifecycler` is not set if it's flush mode, because it need not.